### DISCUSSION TITLE
FEATURE ANNOUNCEMENT: Direct Upload mode in Backups

### DISCUSSION THREAD

**Discussion by ppekrol** [ravendb/ravendb#17679] [Announcements]
Affected versions:
5.4.112+, 6.0.1+

Affected subsystems:
Backups

Beside improving backup performance (https://github.com/ravendb/ravendb/discussions/17678) we decided to also introduce Direct Upload mode to the Backups. This feature allows user to configure a backup in a way that local temp file on disk will not be created, saving you not only disk space, but also putting less pressure on the drive. Drawback of this approach is that only a single destination (for now only AWS S3) can be chosen for backups operating in that mode, but we found that this is usually the case anyway.

We did some testing on our AWS machines in RavenDB Cloud with following results:

| Provider | Instance | Max IOPS | Max throughput | # of Cores | Memory (GB) | Backup Time (Default mode) | Backup Time (Direct Upload mode)
| ------- | ------------------ | ------- | ------------------ |------------------ |------------------ |------------------ |------------------ |
| AWS   | PS10 | 250 | 5 MB/s | 2 | 0.5 | 5:37 | 8:42 |
| AWS   | PS20 | 500 | 10 MB/s | 2 | 1 | 5:17 | 3:50 |
| AWS   | PB10 | 1000 | 21 MB/s | 2 | 2 | 5:18 | 4:06 |
| AWS   | P10 | 3600 | 81 MB/s | 2 | 8 | 5:05 | 3:46 |
| AWS   | P40 | 16000 | 360 MB/s | 16 | 64 | 4:14 | 2:56 |

This feature might be very beneficial for machines with good network bandwidth located near the AWS S3 buckets, but all factors needs to be taken into account, when performance is poor (check PS10), then backups might be taking more time and end up in increased resource utilization (check 'Impact' paragraph here https://github.com/ravendb/ravendb/discussions/17678).

Enjoy!
RavenDB Team

**Comment by sergiomcalzada** [comment]
Waiting for the Azure integration :blush:

**Reply by ppekrol** [reply]
I'm happy to say that this is under development: https://github.com/ravendb/ravendb/pull/17735

**Reply by ppekrol** [reply]
We have released today 5.4.113 & 6.0.2 with Direct Upload to Azure
